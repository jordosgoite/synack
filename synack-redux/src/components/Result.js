import React from 'react';

const Result =({title, descrip, url})=>{
    return(
        <div className="result-container">
            <a href={url} target="_blank" rel="noopener noreferrer">
                <span className="result-url" >{url}</span><span>{title}</span>
            </a>
            <p>{descrip}</p>
        </div>
    )
}
export default Result