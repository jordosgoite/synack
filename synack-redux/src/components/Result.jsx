import React from 'react';
import PropTypes from 'prop-types';

const Result = ({ title, descrip, url }) => (
  <div className="result-container">
    <a href={url} target="_blank" rel="noopener noreferrer">
      <span className="result-url">{url}</span>
      <span>{title}</span>
    </a>
    <p>{descrip}</p>
  </div>
);

Result.propTypes = {
  title: PropTypes.string,
  descrip: PropTypes.string,
  url: PropTypes.string,
};

Result.defaultProps = {
  title: 'test',
  descrip: 'test',
  url: 'test',
};

export default Result;
