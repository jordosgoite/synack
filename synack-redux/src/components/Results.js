import React from 'react';
import { useSelector } from 'react-redux';
import Result from './Result';

const Results =()=>{
    const Search = useSelector((state)=>state.Search)
    const searchEngine = Search.searchEngine;
    return(
        <div>
            {Search.loading && <div className="searching-msj">Searching...</div>}
            {searchEngine==='google' ?
                Search.results.map(nested => nested.map((element, key) =>
                <Result key={key} title={element.title} descrip={element.snippet} url={element.link} />))
                :
                null
            }
            {searchEngine==='bing' ?
                Search.results.map(nested => nested.map((element, key) =>
                <Result key={key} title={element.title} descrip={element.description} url={element.url} />))
                :
                null
            }
            {searchEngine==='both' ?
                <>
                {Search.results[0][0].data.items.map((element, key) =>
                <Result key={key} title={element.title} descrip={element.snippet} url={element.link} />)}
                {Search.results[0][1].data.data.results.organic.map((element, key) =>
                <Result key={key} title={element.title} descrip={element.description} url={element.url} />)}
                </>
                :
                null
            }
            {Search.error && <div>{Search.error}</div>}
        </div>
    )
}
export default Results

