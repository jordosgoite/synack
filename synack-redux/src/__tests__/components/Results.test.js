import React from 'react'
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux'
import configureStore from 'redux-mock-store'
import Results from '../../components/Results.js'
import Result from '../../components/Result.js';

const mockStore = configureStore([]);

describe('React-Redux Results Component', () => {
    let store;
    let component;
   
    beforeEach(() => {
      store = mockStore({
        Search: {
            loading: false, 
            searchEngine:'google', 
            results:[[{title:"test"}, {snippet:"test"},{link:"www.google.com"}]]
        }
      });
   
      component = renderer.create(
        <Provider store={store}>
          <Results />
        </Provider>
      );
    });
   
    it('should render with given state from Redux store', () => {
      expect(component.toJSON()).toMatchSnapshot();
    });
    it('should render the result component', () => {
        renderer.act(() => {
          component.root.find(Result)
        });
      });
  });

