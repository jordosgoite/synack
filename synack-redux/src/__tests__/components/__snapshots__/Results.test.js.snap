// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`React-Redux Results Component should render with given state from Redux store 1`] = `
<div>
  <div
    className="result-container"
  >
    <a
      rel="noopener noreferrer"
      target="_blank"
    >
      <span
        className="result-url"
      />
      <span>
        test
      </span>
    </a>
    <p />
  </div>
  <div
    className="result-container"
  >
    <a
      rel="noopener noreferrer"
      target="_blank"
    >
      <span
        className="result-url"
      />
      <span />
    </a>
    <p>
      test
    </p>
  </div>
  <div
    className="result-container"
  >
    <a
      href="www.google.com"
      rel="noopener noreferrer"
      target="_blank"
    >
      <span
        className="result-url"
      >
        www.google.com
      </span>
      <span />
    </a>
    <p />
  </div>
</div>
`;
